server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -558,6 +559,41 @@ private IndexMetadata getIndexMetadata(String clusterName, String clusterUUID, U
         }
     }
 
+    /**
+     * Fetch global metadata from remote cluster state
+     *
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return {@link IndexMetadata}
+     */
+    public Metadata getGlobalMetadata(String clusterName, String clusterUUID) {
+        start();
+        Optional<ClusterMetadataManifest> clusterMetadataManifest = getLatestClusterMetadataManifest(clusterName, clusterUUID);
```

Comment:
This is called in getLatestIndexMetadata as well. Since this is in download path i am not too worried about the optimizations yet, but this method makes two calls to s3. 

We might need to optimize this later.

Reply:
During Auto Restore, first we would call `getLatestIndexMetadata` and then `getLatestGlobalMetadata` . Each of these methods would call `getLatestClusterMetadataManifest` separately. We need to ensure we wont get different manifest here. Although that should not happen in the happy flow but maybe in an edge case

Reply:
As per my understanding, we are going to restore during quorumloss, at that time there will not be any update in manifest so both should pick same file only. 

But since those are two separate calls to fetch manifest files, there might be some edge case which can cause issue.

Let's do one thing, we will expose method(`getLatestMetadata`) which will return latest Metadata including global metadata and Index Metadata. Caller will take whatever is needed from it. Sounds good?

Reply:
yes 👍